lib/elixir_analyzer/exercise_test/common_checks/function_capture.ex

```diff
@@ -0,0 +1,102 @@
+defmodule ElixirAnalyzer.ExerciseTest.CommonChecks.FunctionCapture do
+  @moduledoc """
+  Check if anonymous functions are used where function capture can be used instead
+  """
+
+  alias ElixirAnalyzer.Constants
+  alias ElixirAnalyzer.Comment
+
+  @spec run(Macro.t()) :: [{:pass | :fail | :skip, %Comment{}}]
+  def run(code_ast) do
+    {_, functions} =
+      Macro.prewalk(code_ast, [], fn ast, functions -> traverse(ast, functions) end)
+
+    case functions |> Enum.map(&format_function(&1)) |> Enum.reverse() do
+      [] ->
+        []
+
+      [{wrong_function, correct_function} | _] ->
+        [
+          {:fail,
+           %Comment{
+             type: :informative,
+             name: Constants.solution_use_function_capture(),
+             comment: Constants.solution_use_function_capture(),
+             params: %{
+               expected: correct_function,
+               actual: wrong_function
+             }
+           }}
+        ]
+    end
+  end
+
+  @exceptions [:<<>>, :{}, :|]
```

Comment:
Something's weird - removing `:{}, :|` from this list still results in the tests passing

Reply:
I shouldn't code too late at night...

For `:{}` I needed to do a test with at least 3 arguments, for 2 arguments the AST looks different.

For `:|` I think I got confused because the following arguments look very similar
```elixir
{:foo, _, [{:a, [], nil}, {:b, [], nil}]} # from fn a, b -> foo(a, b) end
[{:|, _, [{:a, [], Elixir}, {:b, [], Elixir}]}] # from fn a, b -> [a | b]  end
```
The last one is wrapped  in a list, so they are different and I should remove that exception. Thank you!